Sources/macros.js

```diff
@@ -512,7 +512,11 @@ const objectSetterMap = {
   object(publicAPI, model, field) {
     return (value) => {
       if (!DeepEqual(model[field.name], value)) {
-        model[field.name] = value;
+        let _value = value;
+        if (typeof field.process === 'function') {
```

Comment:
I wouldn't bother checking the type of `process`, simply check if it exists. It would be expected to be a function.
Maybe rename `process` with `aboutToBeChanged` or some more explicit naming as such.

Reply:
thank you for your advice